### ISSUE TITLE
AboutOutCmdlets

### ISSUE THREAD

**Issue by arefg** [vexx32/PSKoans#197]
Line 84: Out-File parameter must be FilePath instead of Path

❌ 'Stored knowledge is of little value until it is used.' | Out-File -Path $Path -NoNewline
✔️ 'Stored knowledge is of little value until it is used.' | Out-File -FilePath $Path -NoNewline

Line 125: Apparently there is a carriage return character (`r) in $String variable in addition to every New Line character (`n) in each line which is not the case with here-string so the test will not pass even if I copy and paste the exact output of $String.

         Expected: '\r\n\r\n    Di...'
         But was:  '\n\n        ...'

**Comment by vexx32** [comment]
`-Path` is a valid parameter alias, but you're right I should be using the proper name, as that's present in the help documentation more noticeably.

Ah, that's also a good point. There _is_ that difference... Hm. Not sure how we can fix that, but I'm sure there's a way...

**Comment by arefg** [comment]
About -Path parameter, maybe it's just the case with PowerShell 5.1, because it doesn't recognize the -Path parameter:

                  Out-File : A parameter cannot be found that matches parameter name 'path'.

**Comment by vexx32** [comment]
Oh, they must have fixed that during 6.0 sometime, interesting, I hadn't realized!

**Comment by BrandonLundt** [comment]
> Line 125: Apparently there is a carriage return character (r) in $String variable in addition to every New Line character (n) in each line which is not the case with here-string so the test will not pass even if I copy and paste the exact output of $String.

To me it would be clearer if instead of editing the here string you were editing the statement into | Out-String.

Something like:
```
Describe 'Out-String' {
    <#
        Out-String is a fantastic and simple way to turn complex objects into strings that look just
        like PowerShell's default console formatting.
    #>

    It 'creates string representations of data' {
        $String = ______  | Out-String

        # Fill in this here-string with the data you'd see in your console from calling Get-Item $HOME
        @"

Name                           Value                                                                                                                                                                                         
----                           -----                                                                                                                                                                                         
ColumnA                        Value1                                                                                                                                                                                        
ColumnB                        Value2                                                                                                                                                                                        



"@ | Should -Be $String
        # Mind the indentations; here-strings have to terminate at the very beginning of a line.
    }
}
```

Where the answer is:
```
Describe 'Out-String' {
    <#
        Out-String is a fantastic and simple way to turn complex objects into strings that look just
        like PowerShell's default console formatting.
    #>

    It 'creates string representations of data' {
        $String = @{
            ColumnA = "Value1"
            ColumnB = "Value2"
        } | Out-String

        # Fill in this here-string with the data you'd see in your console from calling Get-Item $HOME
        @"

Name                           Value                                                                                                                                                                                         
----                           -----                                                                                                                                                                                         
ColumnA                        Value1                                                                                                                                                                                        
ColumnB                        Value2                                                                                                                                                                                        



"@ | Should -Be $String
        # Mind the indentations; here-strings have to terminate at the very beginning of a line.
    }
}
```
